### DISCUSSION TITLE
UI.Input() not listening to UI.SameLine()?

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#553] [Q&A] [answered]
Just checking if I'm not missing anything obvious, but with the code below on Windows UWP (HL2) the text input still ends up on the next line, even though it has plenty of window space?

```
UI.WindowBegin("Account", ref window_pose, new Vec2(50, 0) * U.cm, UIWin.Normal);
                UI.Text("Username");
                UI.SameLine();
                UI.Input("username", ref username, new Vec2(15, 0) * U.cm, TextContext.Text);
UI.WindowEnd();
```

**Comment by maluoi** [comment, marked as answer]
`UI.Text` will reserve the whole line automatically! So there's no room left to keep things on the same line. `UI.Label` is probably more appropriate here, as it only reserves enough room for the text provided, but `UI.Text` could work if you use an explicit size.

**Comment by paulmelis** [comment]
Thanks for the hint on difference between `Text` and `Label`, hadn't really noticed that (as I did not check the docs :)). As a follow-up: `UI.Text` does not seem to have a size parameter in the current docs?

Edit: okay, found the size param in the develop branch code!

**Reply by maluoi** [reply]
Yeah, `UI.Text` with size unfortunately got missed in v0.3.6!
